packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,192 @@
+        <span class="accessibility-hidden">Remove</span>
+        <svg width="24" height="24" aria-hidden="true">
+          <use xlink:href="/icons/svg/sprite.svg#close" />
+        </svg>
+      </button>
+    `;
+
+    // Appending node into list
+    this.listElement.appendChild(listElementNode);
+
+    // Appending an item into queue
+    this.fileQueue?.push(file);
+
+    // Adding an event for the delete item button
+    const currentItem = SelectorEngine.findOne(`[data-dismiss="${id}"]`);
+    currentItem?.addEventListener('click', (event) => {
+      event.preventDefault();
+      const target = (event.target as HTMLElement)?.dataset?.dismiss;
+      if (target) {
+        this.removeFromQueue(target);
+      }
+    });
+
+    // Reset drag state
+    this.isDragging = false;
+    this.wrapper?.classList.remove(IS_DRAGGING_CLASS_NAME);
+  }
+
+  addToQueue(file: File) {
+    const allowedFileSize = file.size <= FILE_SIZE_LIMIT;
+    const duplicity = this.fileQueue.find((item) => item.name === file.name);
+    const run = this.fileQueue.length === 0 ? true : this.fileQueue.length >= 1 && !duplicity;
+
+    if (run && allowedFileSize) {
+      this.fileQueue?.push(file);
+      this.appendToList(file);
+    } else if (!allowedFileSize) {
+      console.warn('The file size limit has been exceeded ⚠️');
+    } else {
+      console.warn('This file already exists in the queue ⚠️');
+    }
+  }
+
+  removeFromQueue(name: string) {
+    const index = this.fileQueue.findIndex((item) => item.name.replace(/\.[^/.]+$/, '') === name);
+    if (index > -1) {
+      const element = SelectorEngine.findOne(`li#${name}`);
+      const currentItem = SelectorEngine.findOne(`[data-dismiss="${name}"]`);
+      this.fileQueue.splice(index, 1);
+      currentItem?.removeEventListener('click', () => null);
+      element?.remove();
+    }
+  }
+
+  onChange(event: Event) {
+    const { files } = event.target as HTMLInputElement;
+    const filesArray = files ? [...files] : [];
+
+    filesArray.forEach((file) => this.addToQueue(file));
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  onDragOver(event: Event) {
+    event.preventDefault();
+  }
+
+  onDragEnter() {
+    this.isDragging = true;
+    this.wrapper?.classList.add(IS_DRAGGING_CLASS_NAME);
+  }
+
+  onDragLeave() {
+    this.isDragging = false;
+    this.wrapper?.classList.remove(IS_DRAGGING_CLASS_NAME);
+  }
+
+  onDrop(event: DragEvent) {
+    event.preventDefault();
+
+    const transferItems = Array.from(event?.dataTransfer?.items || []);
+    const transferFiles = Array.from(event?.dataTransfer?.files || []);
+
+    if (transferItems) {
+      transferItems.forEach((item) => {
+        if (item.kind === 'file') {
+          const file = item.getAsFile();
+          if (file) {
+            this.addToQueue(file);
+          }
+        }
+      });
+    } else {
+      transferFiles.forEach((file) => this.addToQueue(file));
+    }
+  }
+
+  initEvents() {
```

Comment:
I prefer naming this method `addEventListeners` as it better describes what does it do, from my point of view.

Reply:
On the contrary, I prefer not to use names identical to global methods

Reply:
Also using `initEvents()` in Collapse component

Reply:
You are right. So I think we should unify this to use naming consistently. There are two methods `initEvents()` right now, counting this PR. And there are three methods `addEventListeners()` (Modal, Offcanvas, Header). I am suggesting using the `addEventListeners` simply because there are more of them. And fix the Collapse later.